### ISSUE TITLE
Not an Issue! A Question

### ISSUE THREAD

**Issue by bud3699** [miek/hd1-unpack#1]
Oki, so first off, I assume by looking at this python script, you dumped the firmware itself, You seem knowledgeable on this radio, therefore I ask a question which you may know the answer to, a while back someone I know tried flashing a firmware upgrade to the Ailunce hd1 radio, however failed, refuses to flash any firmware, booting into dfu mode and attaching to the port manually leads to a flood of the same menu, 1 for update, 2 for view, 3 for execute, booting into update then disconnecting from the port then attaching on the firmware updater, yields it to get to 1%, Ailunce blanked me and stopped responding to my emails when I brought this up to them, so any idea why the menu seems to be looping/flooding with the DFU same menu (I have another model which works fine, expected behaviour is for the menu to show once, the fact its flooding stops anything from being done). Hope you see this, and thank you if you can help

**Comment by miek** [comment]
Hi!
Unfortunately I don't really know too much about the flashing process. I never actually had one of these radios, I was just helping a friend who was reverse-engineering it in the hopes of porting OpenRTX to it. There's a bit of information on the [OpenRTX site](https://openrtx.org/#/hardware/hd1) and this work got integrated into `radio_tool` mentioned there. You may also find someone able to help more on their chat channels.

**Comment by martinling** [comment]
Yes, we were able to dump the firmware using a [hand-written assembly program I wrote](https://github.com/martinling/ailunce-hd1-experiments/) that reads the whole flash out to the UART.

I've not seen the boot loop you describe, but it's possible that it's caused by some corrupted configuration data that's stored in the flash. The firmware update does not overwrite the whole flash. Both the bootloader region at the start, and a configuration area at the end, are left intact.

It might be possible to fix your radio by copying the whole flash contents from another HD1. However, there might be things in there like unit-specific calibration settings adjusted at the factory. So some care would be needed.

The OpenRTX matrix/discord would be a good place to ask for help, all the previous investigation happened there.
